### ISSUE TITLE
Allow pseudo fields to have configurable settings

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#3859]
**Description of the need**
As discussed in https://github.com/backdrop/backdrop-issues/issues/2379, there exists a possible need for pseudo fields to have settings that can be configured per display mode. Pseudo fields can be created to display on the edit and display forms, but unlike real fields on display forms, they cannot be assigned settings that can be configured via a 'Configure' button.

**Proposed solution**
Allow pseudo fields to define settings which can be configured via a 'Configure' button on display mode forms.

**Alternatives that have been considered**
A [workaround](http://www.joshwaihi.com/content/working-extra-fields-drupal-7) exists, but I think it'd be better for core to provide this functionality natively.

**Comment by jenlampton** [comment]
I don't think we should add this API addition without a use in core (and with tests!).  I'd like to propose that this change go in with either https://github.com/backdrop/backdrop-issues/issues/1950 or https://github.com/backdrop/backdrop-issues/issues/1441.

**Comment by BWPanda** [comment]
https://github.com/backdrop/backdrop-issues/issues/4031 is about adding settings to a pseudo field on the Manage Displays form. Whereas https://github.com/backdrop/backdrop-issues/issues/1441 is about adding settings to a pseudo field on the Manage Fields form.

The Manage Displays form's settings are ajax-y. They're like vertical tabs, in that they are shown when you click on them and there's a summary when they're closed/hidden.

The Manage Fields form's settings are separate pages/forms with their own URLs.

So not sure on how best to address both situations with this...
